src/proto/h1/role.rs

```diff
@@ -289,13 +325,41 @@ impl Http1Transaction for Server {
         }))
     }
 
+    #[cfg_attr(feature = "layers",
+        instrument( level = "trace",
+                    skip(msg),
+                    fields( otel.name           = "Server::encode",
+                            otel.kind           = ?SpanKind::Server,
+                            otel.status_code    = ?opentelemetry::trace::StatusCode::Unset,
+                            otel.status_message = tracing::field::Empty,
+                            // OTel required (HTTP)
+                            http.host     = tracing::field::Empty,
+                            http.method   = tracing::field::Empty,
+                            http.scheme   = tracing::field::Empty,
+                            http.target   = tracing::field::Empty,
+                            http.url      = tracing::field::Empty,
+                            net.peer.ip   = tracing::field::Empty,
+                            net.peer.name = tracing::field::Empty,
+                            net.peer.port = tracing::field::Empty,
+                            // OTel optional (HTTP)
+                            http.flavor                 = tracing::field::Empty,
+                            http.request_content_length = tracing::field::Empty,
+                            // OTel optional (General)
+                            net.transport = "IP.TCP",
+                       )
+                    )
+                )]
     fn encode(mut msg: Encode<'_, Self::Outgoing>, dst: &mut Vec<u8>) -> crate::Result<Encoder> {
-        trace!(
-            "Server::encode status={:?}, body={:?}, req_method={:?}",
-            msg.head.subject,
-            msg.body,
-            msg.req_method
-        );
+        let mut otel_status_messages: Vec<String> = vec![];
+        if cfg!(feature = "layers") {
+            tracing::Span::current()
+                .record("http.method", &format!("{:?}", msg.req_method).as_str());
+            tracing::Span::current().record(
+                "http.status_code",
+                &format!("{:?}", msg.head.subject).as_str(),
+            );
+            trace!("Server::encode body={:?}", msg.body);
```

Comment:
prefer structured fields:
```suggestion
            trace!(body = ?msg.body);
```

Reply:
> prefer structured fields:

I'm pretty much leaving existing code unchanged where it is not fully replaced by some span/event structure.  If I was to propose changing this it would be to storing the byte count in one of the span fields.

Reply:
On further reflection I've followed this suggestion - if the proposed approach in the PR gets rejected it won't be because of this :)